### PR TITLE
fix(ci): fix errors in Docker entrypoint changes

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

PR #4409 had a quoting bug.

Our tests should have stopped that PR merging, but we still need to do #4340.

## Solution

- Move shell globs outside quotes
- Print errors and do fallback directory listings
- Limit the length of full directory tree listings
- Remove `exec` and `exit 1` so that we can run commands after `cargo`

## Review

I'm probably going to admin-merge this, because we need some diagnostics before other people start work.

### CODE REVIEW

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -21,52 +21,56 @@ case "$1" in
...
         # For these tests, we activate the gRPC feature to avoid recompiling `zebrad`,
         # but we don't actually run any gRPC tests.
         elif [[ "$TEST_FULL_SYNC" -eq "1" ]]; then
             # Run a Zebra full sync test.
-            exec cargo test --locked --release --features lightwalletd-grpc-tests --package zebrad --test acceptance -- --nocapture --include-ignored full_sync_mainnet
+            cargo test --locked --release --features lightwalletd-grpc-tests --package zebrad --test acceptance -- --nocapture --include-ignored full_sync_mainnet
+            # List directory generated by test
             # TODO: replace with $ZEBRA_CACHED_STATE_DIR in Rust and workflows
-            ls -lh "/zebrad-cache/*/*"
+            ls -lh "/zebrad-cache"/*/* || (echo "No /zebrad-cache/*/*"; ls -lhR "/zebrad-cache" || echo "No /zebrad-cache directory")
```
Comment (dconnolly) [inline docker/entrypoint.sh]: oof

**Review by dconnolly — DISMISSED** [review]

**Review by conradoplg — DISMISSED** [review]
Removing `exec` makes the script fail because it ends in `exit 1`.

To save a bit of time I'll go ahead and remove it just to check if the rest works, but feel free to fix it differently

**Comment by gustavovalverde** [comment]
I think the `ls` command should consider the specific structure required by zebra and lightwalletd cached state for this to work as expected.

This output should fail as it's not the expected structure https://github.com/ZcashFoundation/zebra/runs/6477434066?check_suite_focus=true#step:9:239

P.D.: Seems like @teor2345 were right with their concern on the disk mounting being unreliable. If correct, it's almost certain the workflow will require two separate disks for each cached state (I'll double check this).

**Comment by teor2345** [comment]
> Removing `exec` makes the script fail because it ends in `exit 1`.
> 
> To save a bit of time I'll go ahead and remove it just to check if the rest works, but feel free to fix it differently

This seems to have worked, we're back to the same two failures we were expecting.

**Comment by teor2345** [comment]
> I think the `ls` command should consider the specific structure required by zebra and lightwalletd cached state for this to work as expected.
> 
> This output should fail as it's not the expected structure https://github.com/ZcashFoundation/zebra/runs/6477434066?check_suite_focus=true#step:9:239

The Rust tests already validate the structure, the `ls` command is just designed to list what's there.
(We avoid doing validation in two different places, because then we have to keep them in sync.)

> P.D.: Seems like @teor2345 were right with their concern on the disk mounting being unreliable. If correct, it's almost certain the workflow will require two separate disks for each cached state (I'll double check this).

👍

**Comment by teor2345** [comment]
I admin-merged this PR, so we can get on with merging NU5 mainnet changes.

**Pull request merged**